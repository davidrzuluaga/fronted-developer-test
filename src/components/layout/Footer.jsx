import React from 'react';
import PropTypes from 'prop-types';

const Footer = ({ footerInfo }) => (
  <footer className="text-muted d-none">
    <div className="container text-center">
      <p className="copy">
        { footerInfo.copy }
      </p>
    </div>
  </footer>
);

Footer.propTypes = {
  footerInfo: PropTypes.object
};

export default Footer;
