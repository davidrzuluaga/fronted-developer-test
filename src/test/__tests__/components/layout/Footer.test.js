import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Footer from "../../../../components/layout/Footer";
import projectInfo from "../../../__mocks__/projectInfoMock";

describe('Tests <Footer />', () => {

    const { footerInfo } = projectInfo;

    test('Render component', () => {
        const footer = shallow(<Footer footerInfo={footerInfo}/>);
        expect(footer.length).toEqual(1);
    });

    test('Render copy', () => {
        const footer = mount(<Footer footerInfo={footerInfo}/>);
        expect(footer.find('.copy').text()).toEqual(footerInfo.copy);
    });

    test('Check UI not change', () => {
        const footer = create(<Footer footerInfo={footerInfo}/>);
        expect(footer.toJSON()).toMatchSnapshot();
    });

});
