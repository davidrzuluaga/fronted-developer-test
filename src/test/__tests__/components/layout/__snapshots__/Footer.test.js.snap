// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests <Footer /> Check UI not change 1`] = `
<footer
  className="text-muted fixed-bottom"
>
  <div
    className="container text-center"
  >
    <p
      className="copy"
    >
      Development by Santiago Marulanda
    </p>
  </div>
</footer>
`;
